File: Forums/UserControlCommunication/UserControl1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace UserControlCommunication
{
    public partial class UserControl1 : UserControl
    {
        public delegate void ButtonClickedEvent(string text);
        public ButtonClickedEvent ButtonClicked = null;
        public delegate void SomethingChangedEvent(string text);
        public SomethingChangedEvent SomethingChanged = null;

        public UserControl1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            ButtonClicked?.Invoke("Button clicked at "+DateTime.Now.ToString());
        }

        private void textBox1_Leave(object sender, EventArgs e)
        {
            SomethingChanged?.Invoke(textBox1.Text);
        }
    }
}

File: Forums/UserControlCommunicationVB/UserControl1.vb
Public Class UserControl1
    Public Delegate Sub ButtonClickedEvent(ByVal text As String)
    Public ButtonClicked As ButtonClickedEvent = Nothing
    Public Delegate Sub SomethingChangedEvent(ByVal text As String)
    Public SomethingChanged As SomethingChangedEvent = Nothing

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        ButtonClicked?.Invoke("Button clicked at " & Date.Now.ToString())
    End Sub

    Private Sub TextBox1_Leave(sender As Object, e As EventArgs) Handles TextBox1.Leave
        SomethingChanged?.Invoke(TextBox1.Text)
    End Sub
End Class
